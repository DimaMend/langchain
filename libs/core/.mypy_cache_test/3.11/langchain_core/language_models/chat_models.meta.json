{"data_mtime":1717707325,"dep_lines":[37,57,58,26,27,28,36,38,39,48,55,56,62,63,1,3,4,5,6,7,8,24,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"dep_prios":[5,5,5,5,5,5,5,5,5,5,5,5,25,25,5,10,10,10,10,5,5,5,5,20,20,20,20,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30],"dependencies":["langchain_core.language_models.base","langchain_core.runnables.config","langchain_core.tracers._streaming","langchain_core._api","langchain_core.caches","langchain_core.callbacks","langchain_core.globals","langchain_core.load","langchain_core.messages","langchain_core.outputs","langchain_core.prompt_values","langchain_core.pydantic_v1","langchain_core.runnables","langchain_core.tools","__future__","asyncio","inspect","uuid","warnings","abc","typing","typing_extensions","builtins","pydantic.v1.dataclasses","pydantic.dataclasses","pyexpat.errors","pyexpat.model","_typeshed","asyncio.futures","asyncio.tasks","concurrent","concurrent.futures","concurrent.futures._base","langchain_core._api.deprecation","langchain_core.callbacks.base","langchain_core.callbacks.manager","langchain_core.load.dump","langchain_core.load.serializable","langchain_core.messages.base","langchain_core.outputs.chat_generation","langchain_core.outputs.chat_result","langchain_core.outputs.generation","langchain_core.outputs.llm_result","langchain_core.outputs.run_info","langchain_core.runnables.base","langchain_core.tracers","pydantic","pydantic.v1","pydantic.v1.class_validators","pydantic.v1.fields","pydantic.v1.main","pydantic.v1.utils","types"],"hash":"29a569065805e870d36a267b34794147abf7b10f1f68e3b62137e5265dc222a6","id":"langchain_core.language_models.chat_models","ignore_all":true,"interface_hash":"011bdc0130d02f3a0c945e17ec329db141ca2637b2c970442ba8d23b7935893e","mtime":1717528018,"options":{"allow_redefinition":false,"allow_untyped_globals":false,"always_false":[],"always_true":[],"bazel":false,"check_untyped_defs":false,"disable_bytearray_promotion":false,"disable_error_code":[],"disable_memoryview_promotion":false,"disabled_error_codes":[],"disallow_any_decorated":false,"disallow_any_explicit":false,"disallow_any_expr":false,"disallow_any_generics":false,"disallow_any_unimported":false,"disallow_incomplete_defs":false,"disallow_subclassing_any":false,"disallow_untyped_calls":false,"disallow_untyped_decorators":false,"disallow_untyped_defs":true,"enable_error_code":[],"enabled_error_codes":[],"extra_checks":false,"follow_imports":"normal","follow_imports_for_stubs":false,"ignore_errors":false,"ignore_missing_imports":false,"implicit_optional":false,"implicit_reexport":true,"local_partial_types":false,"mypyc":false,"old_type_inference":false,"platform":"darwin","plugins":[],"strict_concatenate":false,"strict_equality":false,"strict_optional":true,"warn_no_return":true,"warn_return_any":false,"warn_unreachable":false,"warn_unused_ignores":false},"path":"/Users/erickfriis/langchain/oss-py/libs/core/langchain_core/language_models/chat_models.py","plugin_data":null,"size":35556,"suppressed":[],"version_id":"1.10.0"}