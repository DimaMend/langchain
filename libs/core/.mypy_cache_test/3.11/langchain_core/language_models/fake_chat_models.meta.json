{"data_mtime":1717707325,"dep_lines":[11,7,12,13,2,3,4,5,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"dep_prios":[5,5,5,5,10,10,10,5,5,20,20,20,20,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30],"dependencies":["langchain_core.language_models.chat_models","langchain_core.callbacks","langchain_core.messages","langchain_core.outputs","asyncio","re","time","typing","builtins","pydantic.v1.dataclasses","pydantic.dataclasses","pyexpat.errors","pyexpat.model","abc","asyncio.tasks","langchain_core.caches","langchain_core.callbacks.base","langchain_core.callbacks.manager","langchain_core.language_models.base","langchain_core.load","langchain_core.load.serializable","langchain_core.messages.ai","langchain_core.messages.base","langchain_core.messages.tool","langchain_core.outputs.chat_generation","langchain_core.outputs.chat_result","langchain_core.outputs.generation","langchain_core.runnables","langchain_core.runnables.base","pydantic","pydantic.v1","pydantic.v1.main","pydantic.v1.utils"],"hash":"586eda22b5b7f40ae4e59841446a0a43025d34ed858d97165787c30cf19e2937","id":"langchain_core.language_models.fake_chat_models","ignore_all":true,"interface_hash":"21c7761fa3ae6b440136075cd13242d64cfb470394a379f53b7c2ea822a27b83","mtime":1714770705,"options":{"allow_redefinition":false,"allow_untyped_globals":false,"always_false":[],"always_true":[],"bazel":false,"check_untyped_defs":false,"disable_bytearray_promotion":false,"disable_error_code":[],"disable_memoryview_promotion":false,"disabled_error_codes":[],"disallow_any_decorated":false,"disallow_any_explicit":false,"disallow_any_expr":false,"disallow_any_generics":false,"disallow_any_unimported":false,"disallow_incomplete_defs":false,"disallow_subclassing_any":false,"disallow_untyped_calls":false,"disallow_untyped_decorators":false,"disallow_untyped_defs":true,"enable_error_code":[],"enabled_error_codes":[],"extra_checks":false,"follow_imports":"normal","follow_imports_for_stubs":false,"ignore_errors":false,"ignore_missing_imports":false,"implicit_optional":false,"implicit_reexport":true,"local_partial_types":false,"mypyc":false,"old_type_inference":false,"platform":"darwin","plugins":[],"strict_concatenate":false,"strict_equality":false,"strict_optional":true,"warn_no_return":true,"warn_return_any":false,"warn_unreachable":false,"warn_unused_ignores":false},"path":"/Users/erickfriis/langchain/oss-py/libs/core/langchain_core/language_models/fake_chat_models.py","plugin_data":null,"size":10865,"suppressed":[],"version_id":"1.10.0"}