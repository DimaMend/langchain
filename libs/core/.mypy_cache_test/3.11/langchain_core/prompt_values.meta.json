{"data_mtime":1717707325,"dep_lines":[13,14,6,8,9,11,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"dep_prios":[5,5,5,5,5,5,5,20,20,20,20,30,30,30,30,30,30,30,30,30,30,30,30],"dependencies":["langchain_core.load.serializable","langchain_core.messages","__future__","abc","typing","typing_extensions","builtins","pydantic.v1.dataclasses","pydantic.dataclasses","pyexpat.errors","pyexpat.model","langchain_core.load","langchain_core.messages.ai","langchain_core.messages.base","langchain_core.messages.chat","langchain_core.messages.function","langchain_core.messages.human","langchain_core.messages.system","langchain_core.messages.tool","pydantic","pydantic.v1","pydantic.v1.main","pydantic.v1.utils"],"hash":"96789150447c4574089e3f8f4eb07d32cdc131ca956ec6903b576417dbc14f1b","id":"langchain_core.prompt_values","ignore_all":true,"interface_hash":"80243de535e3f754eec00ba1bab7f400a639e3cf66fe110a109e9658b9458434","mtime":1714770705,"options":{"allow_redefinition":false,"allow_untyped_globals":false,"always_false":[],"always_true":[],"bazel":false,"check_untyped_defs":false,"disable_bytearray_promotion":false,"disable_error_code":[],"disable_memoryview_promotion":false,"disabled_error_codes":[],"disallow_any_decorated":false,"disallow_any_explicit":false,"disallow_any_expr":false,"disallow_any_generics":false,"disallow_any_unimported":false,"disallow_incomplete_defs":false,"disallow_subclassing_any":false,"disallow_untyped_calls":false,"disallow_untyped_decorators":false,"disallow_untyped_defs":true,"enable_error_code":[],"enabled_error_codes":[],"extra_checks":false,"follow_imports":"normal","follow_imports_for_stubs":false,"ignore_errors":false,"ignore_missing_imports":false,"implicit_optional":false,"implicit_reexport":true,"local_partial_types":false,"mypyc":false,"old_type_inference":false,"platform":"darwin","plugins":[],"strict_concatenate":false,"strict_equality":false,"strict_optional":true,"warn_no_return":true,"warn_return_any":false,"warn_unreachable":false,"warn_unused_ignores":false},"path":"/Users/erickfriis/langchain/oss-py/libs/core/langchain_core/prompt_values.py","plugin_data":null,"size":3582,"suppressed":[],"version_id":"1.10.0"}