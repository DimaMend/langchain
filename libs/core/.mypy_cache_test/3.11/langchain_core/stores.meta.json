{"data_mtime":1717707325,"dep_lines":[23,24,8,9,1,1,1,1,1,1,1,1,1,1,1],"dep_prios":[5,5,5,5,5,20,20,20,20,30,30,30,30,30,30],"dependencies":["langchain_core.exceptions","langchain_core.runnables","abc","typing","builtins","pydantic.v1.dataclasses","pydantic.dataclasses","pyexpat.errors","pyexpat.model","_collections_abc","_typeshed","concurrent","concurrent.futures","concurrent.futures._base","langchain_core.runnables.config"],"hash":"a82b3ac72727724776f3b4d85955de6b00d0f674271213a54deaf3dbc7ecc5da","id":"langchain_core.stores","ignore_all":true,"interface_hash":"85f2f9110c184be75bf0306bc4305df2c6001caa100afc27293844d516720b87","mtime":1717704699,"options":{"allow_redefinition":false,"allow_untyped_globals":false,"always_false":[],"always_true":[],"bazel":false,"check_untyped_defs":false,"disable_bytearray_promotion":false,"disable_error_code":[],"disable_memoryview_promotion":false,"disabled_error_codes":[],"disallow_any_decorated":false,"disallow_any_explicit":false,"disallow_any_expr":false,"disallow_any_generics":false,"disallow_any_unimported":false,"disallow_incomplete_defs":false,"disallow_subclassing_any":false,"disallow_untyped_calls":false,"disallow_untyped_decorators":false,"disallow_untyped_defs":true,"enable_error_code":[],"enabled_error_codes":[],"extra_checks":false,"follow_imports":"normal","follow_imports_for_stubs":false,"ignore_errors":false,"ignore_missing_imports":false,"implicit_optional":false,"implicit_reexport":true,"local_partial_types":false,"mypyc":false,"old_type_inference":false,"platform":"darwin","plugins":[],"strict_concatenate":false,"strict_equality":false,"strict_optional":true,"warn_no_return":true,"warn_return_any":false,"warn_unreachable":false,"warn_unused_ignores":false},"path":"/Users/erickfriis/langchain/oss-py/libs/core/langchain_core/stores.py","plugin_data":null,"size":8752,"suppressed":[],"version_id":"1.10.0"}