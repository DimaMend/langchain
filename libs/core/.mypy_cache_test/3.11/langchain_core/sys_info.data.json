{".class":"MypyFile","_fullname":"langchain_core.sys_info","future_import_flags":[],"is_partial_stub_package":false,"is_stub":false,"names":{".class":"SymbolTable","Sequence":{".class":"SymbolTableNode","cross_ref":"typing.Sequence","kind":"Gdef"},"__annotations__":{".class":"SymbolTableNode","kind":"Gdef","node":{".class":"Var","flags":["is_ready"],"fullname":"langchain_core.sys_info.__annotations__","name":"__annotations__","type":{".class":"Instance","args":["builtins.str",{".class":"AnyType","missing_import_name":null,"source_any":null,"type_of_any":6}],"type_ref":"builtins.dict"}}},"__doc__":{".class":"SymbolTableNode","kind":"Gdef","node":{".class":"Var","flags":["is_ready"],"fullname":"langchain_core.sys_info.__doc__","name":"__doc__","type":"builtins.str"}},"__file__":{".class":"SymbolTableNode","kind":"Gdef","node":{".class":"Var","flags":["is_ready"],"fullname":"langchain_core.sys_info.__file__","name":"__file__","type":"builtins.str"}},"__name__":{".class":"SymbolTableNode","kind":"Gdef","node":{".class":"Var","flags":["is_ready"],"fullname":"langchain_core.sys_info.__name__","name":"__name__","type":"builtins.str"}},"__package__":{".class":"SymbolTableNode","kind":"Gdef","node":{".class":"Var","flags":["is_ready"],"fullname":"langchain_core.sys_info.__package__","name":"__package__","type":"builtins.str"}},"print_sys_info":{".class":"SymbolTableNode","kind":"Gdef","node":{".class":"FuncDef","abstract_status":0,"arg_kinds":[5],"arg_names":["additional_pkgs"],"dataclass_transform_spec":null,"flags":[],"fullname":"langchain_core.sys_info.print_sys_info","name":"print_sys_info","type":{".class":"CallableType","arg_kinds":[5],"arg_names":["additional_pkgs"],"arg_types":[{".class":"Instance","args":["builtins.str"],"type_ref":"typing.Sequence"}],"bound_args":[],"def_extras":{"first_arg":null},"fallback":"builtins.function","from_concatenate":false,"implicit":false,"imprecise_arg_kinds":false,"is_ellipsis_args":false,"name":"print_sys_info","ret_type":{".class":"NoneType"},"type_guard":null,"type_is":null,"unpack_kwargs":false,"variables":[]}}}},"path":"/Users/erickfriis/langchain/oss-py/libs/core/langchain_core/sys_info.py"}