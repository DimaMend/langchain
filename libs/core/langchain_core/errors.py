from enum import Enum


class ErrorCode(Enum):
    INVALID_PROMPT_INPUT = "INVALID_PROMPT_INPUT"
    INVALID_TOOL_RESULTS = "INVALID_TOOL_RESULTS"
    MESSAGE_COERCION_FAILURE = "MESSAGE_COERCION_FAILURE"
    MODEL_AUTHENTICATION = "MODEL_AUTHENTICATION"
    MODEL_NOT_FOUND = "MODEL_NOT_FOUND"
    MODEL_RATE_LIMIT = "MODEL_RATE_LIMIT"
    OUTPUT_PARSING_FAILURE = "OUTPUT_PARSING_FAILURE"


def create_message(*, message: str, error_code: ErrorCode) -> str:
    return (
        f"{message}\n"
        "For troubleshooting, visit: https://python.langchain.com/docs/"
        f"troubleshooting/errors/{error_code.value}"
    )
