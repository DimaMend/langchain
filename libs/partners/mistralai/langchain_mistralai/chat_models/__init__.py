from langchain_mistralai.chat_models.base import ChatMistralAI
from langchain_mistralai.chat_models.azure import AzureChatMistralAI

__all__ = ["ChatMistralAI", "AzureChatMistralAI"]
