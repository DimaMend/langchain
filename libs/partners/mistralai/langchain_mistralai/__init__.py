from langchain_mistralai.chat_models import (
    ChatMistralAI,
    AzureChatMistralAI,
)
from langchain_mistralai.embeddings import MistralAIEmbeddings

__all__ = ["ChatMistralAI", "MistralAIEmbeddings", "AzureChatMistralAI"]
