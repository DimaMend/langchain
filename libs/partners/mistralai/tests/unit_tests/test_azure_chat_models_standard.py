"""Standard LangChain interface tests"""

from typing import Tuple, Type

import pytest
from langchain_core.language_models import BaseChatModel
from langchain_standard_tests.unit_tests import ChatModelUnitTests

from langchain_mistralai import AzureChatMistralAI


class TestAzureMistralAIStandard(ChatModelUnitTests):
    @property
    def chat_model_class(self) -> Type[BaseChatModel]:
        return AzureChatMistralAI

    @property
    def chat_model_params(self) -> dict:
        return {
            "azure_endpoint": "https://test.azure.com",
        }

    @pytest.mark.xfail(reason="Azure does not support tool_choice='any'")
    def test_bind_tool_pydantic(self, model: BaseChatModel) -> None:
        super().test_bind_tool_pydantic(model)

    @property
    def init_from_env_params(self) -> Tuple[dict, dict, dict]:
        return (
            {
                "AZURE_MISTRAL_ENDPOINT": "https://azure.mistral-endpoint.com/",
                "AZURE_MISTRAL_API_KEY": "api_key",
            },
            {},
            {
                "mistral_api_key": "api_key",
                "azure_endpoint": "https://azure.mistral-endpoint.com/",
            },
        )
