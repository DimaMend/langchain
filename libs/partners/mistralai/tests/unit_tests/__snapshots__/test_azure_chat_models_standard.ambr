# serializer version: 1
# name: TestAzureMistralAIStandard.test_serdes[serialized]
  dict({
    'id': list([
      'langchain_mistralai',
      'chat_models',
      'azure',
      'AzureChatMistralAI',
    ]),
    'kwargs': dict({
      'azure_endpoint': 'https://test.azure.com',
      'endpoint': 'https://test.azure.com',
      'max_concurrent_requests': 64,
      'max_retries': 2,
      'max_tokens': 100,
      'mistral_api_key': dict({
        'id': list([
          'MISTRAL_API_KEY',
        ]),
        'lc': 1,
        'type': 'secret',
      }),
      'model': 'mistral-small',
      'temperature': 0.0,
      'timeout': 60,
      'top_p': 1,
    }),
    'lc': 1,
    'name': 'AzureChatMistralAI',
    'type': 'constructor',
  })
# ---
